### PR TITLE
[EDU-2448] refactor: cache keys into edge caching

### PR DESCRIPTION
**Pull request by bru-andrade**
### Changes

- Removed cache key format information from RTP
- Added cache key section to edge caching page

### Additional links

[EDU-3000]: https://aziontech.atlassian.net/browse/EDU-3000?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[EDU-2448]: https://aziontech.atlassian.net/browse/EDU-2448?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: src/content/docs/en/pages/main-menu/reference/build/edge-application/edge-caching.mdx [hunk 1]
```diff
@@ -27,6 +27,78 @@ With Azion, you can customize the time the cache remains in the edge through *ti
 
 ---
 
+## Cache keys
+
+A cache key is an index entry for an object in Azion's edge nodes. When a first request is made to an edge application, the edge requests the objects within the request to the origin server. This generates a cache for each object at the edge. When the next request for the same resource is made to the edge application, if the resource requested matches one already found through the cache key, the request won't be forwarded to the origin.
```
Comment (bru-andrade) [inline src/content/docs/en/pages/main-menu/reference/build/edge-application/edge-caching.mdx:32]:
```suggestion
A cache key is an index entry for an object in Azion's edge nodes. When a request is made to an edge application for the first time, the edge requests the objects within the request to the origin server. This generates a cache for each object at the edge. When the next request for the same resource is made to the edge application, if the resource requested matches one already found through the cache key, the request won't be forwarded to the origin.
```

File: src/content/docs/en/pages/main-menu/reference/build/edge-application/edge-caching.mdx [hunk 2]
```diff
@@ -27,6 +27,85 @@ With Azion, you can customize the time the cache remains in the edge through *ti
 
 ---
 
+## Cache keys
+
+A cache key is an index entry for an object in Azion's edge nodes. When a request is made to an edge application for the first time, the edge forwards the request to the origin server and generates a cache for each object received at the edge. When the next request for the same resource is made to the edge application, if the resource requested matches one already found through the cache key, the request won't be forwarded to the origin.
+
+You can obtain the cache key of an application's resource through a `HEAD` request for the content URI and add the Azion debug header `Pragma: azion-debug-cache`:
```
Comment (arturdj) [inline src/content/docs/en/pages/main-menu/reference/build/edge-application/edge-caching.mdx:34]: The request doesn't have to be of HEAD method. Any method (GET, OPTIONS, etc) does return pragma headers. The only exception is when the return is an error (4xx, 5xx)

File: src/content/docs/en/pages/main-menu/reference/build/edge-application/edge-caching.mdx [hunk 3]
```diff
@@ -27,6 +27,85 @@ With Azion, you can customize the time the cache remains in the edge through *ti
...
+
+```bash
+curl --location --head 'https://yourdomain.com/resource-path/image.jpeg' \
```
Comment (arturdj) [inline src/content/docs/en/pages/main-menu/reference/build/edge-application/edge-caching.mdx:37]: --head is not mandatory.

**Review by MarianaAguilera — APPROVED** [review]

**Review by Leticia-JRosa — APPROVED** [review]
Approved!!

**Review by hannah-sk — APPROVED** [review]

File: src/content/docs/en/pages/main-menu/reference/build/edge-application/edge-caching.mdx (hunk 2 above)
Comment (bru-andrade) [inline src/content/docs/en/pages/main-menu/reference/build/edge-application/edge-caching.mdx:34]:
```suggestion
You can obtain the cache key of an application's resource through a request for the content URI and add the Azion debug header `Pragma: azion-debug-cache`:
```

File: src/content/docs/en/pages/main-menu/reference/build/edge-application/edge-caching.mdx [hunk 4]
```diff
@@ -27,6 +27,85 @@ With Azion, you can customize the time the cache remains in the edge through *ti
...
+A cache key is an index entry for an object in Azion's edge nodes. When a request is made to an edge application for the first time, the edge forwards the request to the origin server and generates a cache for each object received at the edge. When the next request for the same resource is made to the edge application, if the resource requested matches one already found through the cache key, the request won't be forwarded to the origin.
+
+You can obtain the cache key of an application's resource through a request for the content URI and add the Azion debug header `Pragma: azion-debug-cache`:
+
+```bash
+curl --location --head 'https://yourdomain.com/resource-path/image.jpeg' \
+--header 'Pragma: azion-debug-cache'
+```
+
+The response will return an `X-Cache-Key` header with the cache key value, which based on the example above, would be:
```
Comment (bru-andrade) [inline src/content/docs/en/pages/main-menu/reference/build/edge-application/edge-caching.mdx:41]:
```suggestion
A successful response will return an `X-Cache-Key` header with the cache key value, which based on the example above, would be:
```

File: src/content/docs/en/pages/main-menu/reference/build/edge-application/edge-caching.mdx (hunk 3 above)
Comment (bru-andrade) [reply src/content/docs/en/pages/main-menu/reference/build/edge-application/edge-caching.mdx:37]: Simplified the command and explanation 😉

File: src/content/docs/en/pages/main-menu/reference/build/edge-application/edge-caching.mdx [hunk 5]
```diff
@@ -27,6 +27,85 @@ With Azion, you can customize the time the cache remains in the edge through *ti
...
 
+## Cache keys
+
+A cache key is an index entry for an object in Azion's edge nodes. When a request is made to an edge application for the first time, the edge forwards the request to the origin server and generates a cache for each object received at the edge. When the next request for the same resource is made to the edge application, if the resource requested matches one already found through the cache key, the request won't be forwarded to the origin.
+
+You can obtain the cache key of an application's resource through a request for the content URI and add the Azion debug header `Pragma: azion-debug-cache`:
+
+```bash
+curl --location --head 'https://yourdomain.com/resource-path/image.jpeg' \
+--header 'Pragma: azion-debug-cache'
```
Comment (bru-andrade) [inline src/content/docs/en/pages/main-menu/reference/build/edge-application/edge-caching.mdx:38]:
```suggestion
curl -I https://yourdomain.com/resource-path/image.jpeg -H "Pragma: azion-debug-cache"
```

File: src/content/docs/en/pages/main-menu/reference/build/edge-application/edge-caching.mdx [hunk 6]
```diff
@@ -27,6 +27,84 @@ With Azion, you can customize the time the cache remains in the edge through *ti
...
+
+:::note
+Find out more about how to obtain other cache data in the [How to verify application cache indicators using Modheader for Google Chrome](/en/documentation/products/guides/check-page-cache-time/) guide.
+:::
+
+### Cache key format
+
+The default cache key format adopted by Azion concatenates the following URI syntax elements:
+
+- Scheme of the request
```
Comment (bru-andrade) [inline src/content/docs/en/pages/main-menu/reference/build/edge-application/edge-caching.mdx:54]:
```suggestion
- Scheme of the request (for complex requests)
```

File: src/content/docs/en/pages/main-menu/reference/build/edge-application/edge-caching.mdx [hunk 7]
```diff
@@ -27,6 +27,84 @@ With Azion, you can customize the time the cache remains in the edge through *ti
...
+- Host
+- Path to the resource
+- Query string
```
Comment (bru-andrade) [inline src/content/docs/en/pages/main-menu/reference/build/edge-application/edge-caching.mdx:57]:
```suggestion
- Query string
- A variation separator and appended variations
```

File: src/content/docs/en/pages/main-menu/reference/build/edge-application/real-time-purge.mdx [hunk 8]
```diff
@@ -170,28 +126,16 @@ You can't purge from the **L2 Caching** layer using a wildcard expression.
 
 You can purge objects with content variation based on cookies or query strings using the **Advanced Cache Key** feature from the [Application Acceleration](/en/documentation/products/edge-application/application-acceleration/) module. 
 
-In addition to `host` and `path`, the cache key must include the separator `@@` followed by the name of the cookies used and their values. This type of cache key ends in `;`.
 
 To remove objects with cookie-based variation, you can use a purge by cache key, entering all variations individually, or a purge by wildcard, using `@@*` at the end.
```
Comment (bru-andrade) [inline src/content/docs/en/pages/main-menu/reference/build/edge-application/real-time-purge.mdx:130]:
```suggestion
To remove objects with [cookie-based variation](/en/documentation/products/edge-application/edge-caching/#advanced-cache-key), you can use a purge by cache key, entering all variations individually, or a purge by wildcard, using `@@*` at the end.
```

File: src/content/docs/en/pages/main-menu/reference/build/edge-application/real-time-purge.mdx [hunk 9]
```diff
@@ -170,28 +126,16 @@ You can't purge from the **L2 Caching** layer using a wildcard expression.
...
-- `httpwww.yourdomain.com/@@;`
-- `httpwww.yourdomain.com/@@user=user1;`
-- `httpwww.yourdomain.com/@@user=user2;`
-
-When using query string based content variation, in addition to the `host` and `path`, the cache key must include the `?` separator and the query string arguments used. For example:
-
-- `httpsdynamic.yourdomain.com/product.py?id=1000`
-- `httpsdynamic.yourdomain.com/product.py?id=1001`
-
 To purge objects with variation based on query string, you can use:
```
Comment (bru-andrade) [inline src/content/docs/en/pages/main-menu/reference/build/edge-application/real-time-purge.mdx:132]:
```suggestion
To purge objects with [variation based on query string](/en/documentation/products/edge-application/edge-caching/#advanced-cache-key), you can use: 
```

File: src/content/docs/en/pages/main-menu/reference/build/edge-application/edge-caching.mdx [hunk 10]
```diff
@@ -27,6 +27,85 @@ With Azion, you can customize the time the cache remains in the edge through *ti
...
+
+:::note
+Find out more about how to obtain other cache data in the [How to verify application cache indicators using Modheader for Google Chrome](/en/documentation/products/guides/check-page-cache-time/) guide.
+:::
+
+### Cache key format
+
+The default cache key format adopted by Azion concatenates the following URI syntax elements:
+
+- Scheme of the request (for complex requests)
```
Comment (bru-andrade) [inline src/content/docs/en/pages/main-menu/reference/build/edge-application/edge-caching.mdx:54]:
```suggestion
- Scheme of the request
```

File: src/content/docs/en/pages/main-menu/reference/build/edge-application/edge-caching.mdx [hunk 11]
```diff
@@ -27,6 +27,85 @@ With Azion, you can customize the time the cache remains in the edge through *ti
...
+:::
+
+### Cache key format
+
+The default cache key format adopted by Azion concatenates the following URI syntax elements:
+
+- Scheme of the request
+- Host
+- Path to the resource
+- Query string
```
Comment (bru-andrade) [inline src/content/docs/en/pages/main-menu/reference/build/edge-application/edge-caching.mdx:57]:
```suggestion
- Query strings
```

File: src/content/docs/en/pages/main-menu/reference/build/edge-application/edge-caching.mdx [hunk 12]
```diff
@@ -27,6 +27,85 @@ With Azion, you can customize the time the cache remains in the edge through *ti
...
+
+### Cache key format
+
+The default cache key format adopted by Azion concatenates the following URI syntax elements:
+
+- Scheme of the request
+- Host
+- Path to the resource
+- Query strings
+- A variation separator and appended variations
```
Comment (bru-andrade) [inline src/content/docs/en/pages/main-menu/reference/build/edge-application/edge-caching.mdx:58]:
```suggestion
- A variation separator and, when implemented, appended variations
```

File: src/content/docs/en/pages/main-menu/reference/build/edge-application/edge-caching.mdx [hunk 13]
```diff
@@ -27,6 +27,85 @@ With Azion, you can customize the time the cache remains in the edge through *ti
...
+
+- Scheme of the request
+- Host
+- Path to the resource
+- Query strings
+- A variation separator and, when implemented, appended variations
+
+An object with the URI `https://static.yourdomain.com/page/site.js` should generate the cache key `httpsstatic.yourdomain.com/page/site.js`.
+
+Some cache keys may also change depending on the following variations generated at the edge.
```
Comment (bru-andrade) [inline src/content/docs/en/pages/main-menu/reference/build/edge-application/edge-caching.mdx:62]:
```suggestion
Some cache keys may also change depending on variations generated at the edge. If variations are enabled, the cache key may contain the `@@` separator at the end, resulting in `httpsstatic.seudominio.com/pagina/site.js@@`.
```

File: src/content/docs/en/pages/main-menu/reference/build/edge-application/edge-caching.mdx [hunk 14]
```diff
@@ -27,6 +27,85 @@ With Azion, you can customize the time the cache remains in the edge through *ti
...
+- Query strings
+- A variation separator and, when implemented, appended variations
+
+An object with the URI `https://static.yourdomain.com/page/site.js` should generate the cache key `httpsstatic.yourdomain.com/page/site.js`.
+
+Some cache keys may also change depending on variations generated at the edge. If variations are enabled, the cache key may contain the `@@` separator at the end, resulting in `httpsstatic.seudominio.com/pagina/site.js@@`.
+
+#### Complex requests
+
+If the resource is obtained using a complex request (any method different than `GET` or `HEAD`), the method is added to the cache key. For example, for an `OPTIONS` request, the cache key generated for an object would be `optionshttpsstatic.yourdomain.com/page`.
```
Comment (bru-andrade) [inline src/content/docs/en/pages/main-menu/reference/build/edge-application/edge-caching.mdx:66]:
```suggestion
If the resource is obtained using a complex request (any method different than `GET` or `HEAD`), the method is added to the start of the cache key.

For example, for an `OPTIONS` request, the cache key generated for an object would be `optionshttpsstatic.yourdomain.com/page`.
```

File: src/content/docs/en/pages/main-menu/reference/build/edge-application/edge-caching.mdx [hunk 15]
```diff
@@ -27,6 +27,85 @@ With Azion, you can customize the time the cache remains in the edge through *ti
...
+
+#### Image Processor
+
+When [Image Processor](/en/documentation/products/edge-application/image-processor/) is activated and an image is resized and converted into `webp` format, the cache key generated will be `httpsstatic.yourdomain.com/static/images/image_1.jpg?ims=880x@@webp`.
```
Comment (bru-andrade) [inline src/content/docs/en/pages/main-menu/reference/build/edge-application/edge-caching.mdx:70]:
```suggestion
When [Image Processor](/en/documentation/products/edge-application/image-processor/) is activated, each `ims` query element is separated by the separator.

For example, for an image that is resized and converted into `webp` format, the cache key generated will be `httpsstatic.yourdomain.com/static/images/image_1.jpg?ims=880x@@webp`.
```

File: src/content/docs/en/pages/main-menu/reference/build/edge-application/edge-caching.mdx [hunk 16]
```diff
@@ -27,6 +27,89 @@ With Azion, you can customize the time the cache remains in the edge through *ti
...
+#### File slicing
+
+For sliced objects, the cache key appends the `@@bytes=` separator for each slice of content. Cache keys using optimization for delivering large files have the following format:
+
+- `httpstatic.yourdomain.com/media/file.mp4@@bytes=0-1048575`
+- `httpstatic.yourdomain.com/media/file.mp4@@bytes=1048576-2097151`
+
+#### Cached HTTP methods
+
+For cached `POST` or `OPTIONS` requests, the cache key appends the `@@` separator followed by the MD5 hash of the request body. For example:
```
Comment (bru-andrade) [inline src/content/docs/en/pages/main-menu/reference/build/edge-application/edge-caching.mdx:106]:
```suggestion
For cached `POST` or `OPTIONS` requests, the cache key appends the `@@` separator followed by the MD5 hash of the request body. 

For example:
```

File: src/content/docs/en/pages/main-menu/reference/build/edge-application/edge-caching.mdx [hunk 17]
```diff
@@ -27,6 +27,89 @@ With Azion, you can customize the time the cache remains in the edge through *ti
...
+
+- `httpstatic.yourdomain.com/page?name=name`
+- `httpstatic.yourdomain.com/page?city=city&name=name`
+- `httpstatic.yourdomain.com/page?name=name&city=city`
+
+For multiple query strings, if **Query string sort** is enabled, the query strings will be grouped under the same cache key, ordered alphabetically.
+
+#### File slicing
+
+For sliced objects, the cache key appends the `@@bytes=` separator for each slice of content. Cache keys using optimization for delivering large files have the following format:
```
Comment (bru-andrade) [inline src/content/docs/en/pages/main-menu/reference/build/edge-application/edge-caching.mdx:99]:
```suggestion
For sliced objects, the cache key appends the `@@bytes=` separator for each slice of content.

For example, for a file that is `2097151` bytes in size, if file slicing is enabled, the cache keys generated will be:
```

File: src/content/docs/en/pages/main-menu/reference/build/edge-application/edge-caching.mdx [hunk 18]
```diff
@@ -27,6 +27,89 @@ With Azion, you can customize the time the cache remains in the edge through *ti
...
+#### Advanced Cache Key
+
+When the [Advanced Cache Key](/en/documentation/products/edge-application/application-acceleration/#advanced-cache-key) feature is configured, the formation of cache keys is also affected. 
+
+**Cache by Cookie** generates a cache key variation with the determined cookie names and values followed by `;`. For example, content variation based on the `user` cookie could generate the following cache keys:
+
+- `httpwww.yourdomain.com/@@;`
+- `httpwww.yourdomain.com/@@user=user;`
+
+**Cache by Query String** generates cache keys with the query separator and the determined arguments based on the order of the query strings submitted in the request. For example, content variation based on `name` and `city` could generate the following cache keys:
```
Comment (bru-andrade) [inline src/content/docs/en/pages/main-menu/reference/build/edge-application/edge-caching.mdx:89]:
```suggestion
**Cache by Query String** generates cache keys with the query separator and the determined arguments based on the order of the query strings submitted in the request. 

For example, content variation based on `name` and `city` could generate the following cache keys:
```

File: src/content/docs/en/pages/main-menu/reference/build/edge-application/edge-caching.mdx [hunk 19]
```diff
@@ -27,6 +27,89 @@ With Azion, you can customize the time the cache remains in the edge through *ti
...
+
+#### Device Groups
+
+If there's an object variation per [device group](/en/documentation/products/edge-application/device-groups/), the cache key will be appended with the separator `@@` and the device group name; for example: `httpwww.yourdomain.com/@@Mobile`.
+
+#### Advanced Cache Key
+
+When the [Advanced Cache Key](/en/documentation/products/edge-application/application-acceleration/#advanced-cache-key) feature is configured, the formation of cache keys is also affected. 
+
+**Cache by Cookie** generates a cache key variation with the determined cookie names and values followed by `;`. For example, content variation based on the `user` cookie could generate the following cache keys:
```
Comment (bru-andrade) [inline src/content/docs/en/pages/main-menu/reference/build/edge-application/edge-caching.mdx:84]:
```suggestion
**Cache by Cookie** generates a cache key variation with the determined cookie names and values followed by `;`. 

For example, content variation based on the `user` cookie could generate the following cache keys:
```

File: src/content/docs/en/pages/main-menu/reference/build/edge-application/edge-caching.mdx [hunk 20]
```diff
@@ -27,6 +27,89 @@ With Azion, you can customize the time the cache remains in the edge through *ti
...
+
+#### Image Processor
+
+When [Image Processor](/en/documentation/products/edge-application/image-processor/) is activated, each `ims` query element is separated by the separator.
+
+For example, for an image that is resized and converted into `webp` format, the cache key generated will be `httpsstatic.yourdomain.com/static/images/image_1.jpg?ims=880x@@webp`.
+
+#### Device Groups
+
+If there's an object variation per [device group](/en/documentation/products/edge-application/device-groups/), the cache key will be appended with the separator `@@` and the device group name; for example: `httpwww.yourdomain.com/@@Mobile`.
```
Comment (bru-andrade) [inline src/content/docs/en/pages/main-menu/reference/build/edge-application/edge-caching.mdx:78]:
```suggestion
If there's an object variation per [device group](/en/documentation/products/edge-application/device-groups/), the cache key will be appended with the separator `@@` and the device group name

For example, for an application that implements variation for the `Mobile` group, the generated cache key will be `httpwww.yourdomain.com/@@Mobile`.
```

File: src/content/docs/en/pages/main-menu/reference/build/edge-application/edge-caching.mdx [hunk 21]
```diff
@@ -27,6 +27,89 @@ With Azion, you can customize the time the cache remains in the edge through *ti
...
+
+#### Complex requests
+
+If the resource is obtained using a complex request (any method different than `GET` or `HEAD`), the method is added to the start of the cache key.
+
+For example, for an `OPTIONS` request, the cache key generated for an object would be `optionshttpsstatic.yourdomain.com/page`.
+
+#### Image Processor
+
+When [Image Processor](/en/documentation/products/edge-application/image-processor/) is activated, each `ims` query element is separated by the separator.
```
Comment (bru-andrade) [inline src/content/docs/en/pages/main-menu/reference/build/edge-application/edge-caching.mdx:72]:
```suggestion
When [Image Processor](/en/documentation/products/edge-application/image-processor/) is activated, in addition to the `ims` query string, variations with converted format contain the file format after the separator.
```

File: src/content/docs/en/pages/main-menu/reference/build/edge-application/real-time-purge.mdx [hunk 22]
```diff
@@ -214,11 +158,6 @@ You can use a cache key purge, entering all variations individually, or a wildca
 
 If you use the [Slice](/en/documentation/products/edge-application/cache-settings/#slice-settings) feature in your application, large files might still remain in cache.
```
Comment (bru-andrade) [inline src/content/docs/en/pages/main-menu/reference/build/edge-application/real-time-purge.mdx:159]:
```suggestion
If you use the [Slice](/en/documentation/products/edge-application/cache-settings/#slice-settings) feature in your application, large files might still remain in cache. To perform a purge in this case, you must list the cache keys for each slice individually.
```

File: src/content/docs/en/pages/main-menu/reference/build/edge-application/edge-caching.mdx (hunk 2 above)
Comment (bru-andrade) [reply src/content/docs/en/pages/main-menu/reference/build/edge-application/edge-caching.mdx]: Left general statement!

**Pull request merged**